component-catalog-connectors/artifactory-connector/README.md

```diff
@@ -0,0 +1,203 @@
+repository_name      = elyra-components
+repository_path      = /
+max_recursion_depth  = 3
+max_files_per_folder = -1
+file_filter          = *.yaml
+file_ordering        = VERSION_DESCENDING
+
+Matched:
+=========
+https://example.com/artifactory/elyra-components/component_1/component-1.0.9.yaml
+https://example.com/artifactory/elyra-components/component_1/component-1.0.10.yaml
+https://example.com/artifactory/elyra-components/component_2/component-1.0.0.yaml
+https://example.com/artifactory/elyra-components/component_2/component-1.1.0.yaml
+
+Notes:
+=========
+- the `component_3/` files are not matched as this folder does not contain a `__COMPONENT__` marker
+- the `component_2/hidden_component/` files are not matched as recursion stops at the first `__COMPONENT__` marker
+```
+
+__Example 2:__
+
+```
+Configs:
+=========
+artifactory_url      = https://example.com/artifactory/
+repository_name      = elyra-components
+repository_path      = /
+max_recursion_depth  = 0
+max_files_per_folder = -1
+file_filter          = *.yaml
+file_ordering        = VERSION_DESCENDING
+
+Matched:
+=========
+N/A
+
+Notes:
+=========
+- no files are matched, as `max_recursion_depth` is `0`
+```
+
+__Example 3:__
+
+```
+Configs:
+=========
+artifactory_url      = https://example.com/artifactory/
+repository_name      = elyra-components
+repository_path      = /
+max_recursion_depth  = 3
+max_files_per_folder = 1
+file_filter          = *.yaml
+file_ordering        = VERSION_DESCENDING
+
+Matched:
+=========
+https://example.com/artifactory/elyra-components/component_1/component-1.0.10.yaml
+https://example.com/artifactory/elyra-components/component_2/component-1.1.0.yaml
+
+Notes:
+=========
+- the `file_ordering` is applied separately within each folder
+- as `max_files_per_folder` is `1`, only ONE file from each folder is matched 
+- as `file_ordering` is `VERSION_DESCENDING`, the file names are ordered as if they are version numbers
+  (we use `packaging.version.LegacyVersion()` to preform the sort)
+- the whole file-name is treated as a version, so "aaaa-1.0.0.yaml" is sorted before "bbbb-9.0.0.yaml"
+  (take care not to change your file-name prefixes, or alternatively don't include a prefix and use "1.0.0.ymal")
+```
+
+## Uninstall the connector
+
+1. Remove all Artifactory catalog entries from the `Manage Components` panel
+2. Stop JupyterLab
+3. Uninstall the `elyra-artifactory-catalog-connector` package:
+   ```
+   $ pip uninstall elyra-artifactory-catalog-connector
+   ```
+
+## Troubleshooting
+
+- __Problem:__ The palette does not display any components from the configured catalog.
+   - __Solution:__ If the Elyra GUI does not display any error message indicating that a problem was encountered, inspect the JupyterLab log file.
+- __Problem:__ The pallet does not reflect the current state of the Artifactory repo.
+   - __Solution:__ Trigger the catalog to refresh by editing the catalog, and clicking "SAVE & CLOSE" (without making any changes).
```

Comment:
We've added the ability to  reload the content of a single catalog or all catalogs in [version 3.7](https://github.com/elyra-ai/elyra/releases/tag/v3.7.0). Therefore there should be no longer a need for this workaround.